crates/stages/src/stages/bodies.rs

```diff
@@ -172,7 +167,7 @@ impl<DB: Database, D: BodyDownloader, C: Consensus> Stage<DB> for BodyStage<D, C
         // The stage is "done" if:
         // - We got fewer blocks than our target
         // - We reached our target and the target was not limited by the batch size of the stage
-        let done = !capped && highest_block == end_block;
+        let done = highest_block == end_block;
```

Comment:
This is only correct if `end_block` represents the full range and is not capped by commit threshold. If it's capped by commit threshold, then the old check should be added back.

If `end_block` represents the full range then the comment above should be updated to reflect it.

Reply:
end_block is now not affecting the commit threshold here because the stream yields us only the desired number of blocks, whereas before we were mutating end_block